lib/metasploit/framework/data_service/remote/http/core.rb

```diff
@@ -24,15 +24,25 @@ class RemoteHTTPDataService
   #
   # @param [String] endpoint A valid http or https URL. Cannot be nil
   #
-  def initialize(endpoint, framework, https_opts = {})
-    @active = false
+  def initialize(endpoint, https_opts = {})
     validate_endpoint(endpoint)
     @endpoint = URI.parse(endpoint)
     @https_opts = https_opts
-    @framework = framework
     build_client_pool(5)
   end
 
+  def online?
```

Comment:
Why implement another online check? This was already implemented via the [`active`](https://github.com/rapid7/metasploit-framework/blob/master/lib/metasploit/framework/data_service/remote/http/core.rb#L48) and [`is_online?`](https://github.com/rapid7/metasploit-framework/blob/master/lib/metasploit/framework/data_service/remote/http/core.rb#L237) methods in #9926.

Reply:
ha! didn't notice, probably because the is_online? object was private